### DISCUSSION TITLE
How to export only a portion of the Model space elements to pdf file?

### DISCUSSION THREAD

**Discussion by maslke** [mozman/ezdxf#1066] [Q&A] [answered]
Hello everyone.
 I have a question I'd like to ask. I have a DWG file as shown below. I already know how to export all the elements in the entire Model space to a PDF file. Now I'm wondering if there's a way to export only a portion of the Model space elements. In this sample DWG file, it's about exporting the elements within the first frame only.
![image](https://github.com/mozman/ezdxf/assets/6869592/555ef6e7-66f2-4a63-8d03-ac10d9a043fa)

Thank you.
[template.zip](https://github.com/mozman/ezdxf/files/15036342/template.zip)

**Comment by mozman** [comment, marked as answer]
You don't have to draw the whole modelspace, you can also draw just a subset of the entities: `Frontend.draw_entities(subset)`

How to get this subset? That is your specific problem to solve. The next version of `ezdxf` has a new `select` module to select entities based on their location. A release candidate `ezdxf v1.3.0rc0` is available on PyPI.

Docs for the  selection tools: https://ezdxf.mozman.at/docs/tasks/select.html
Tutorial: https://ezdxf.mozman.at/docs/tutorials/entity_selection.html#tut-entity-selection

**Reply by maslke** [reply]
Thank you for your response. 
I have just resolved my issue through other ways. The general method was: 
1. Create a Paperspace. 
3. use add_viewport method.

The code is just like this:
```python
start = [13225861.9385, 3755977.9181]
page_width = 425
for i in range(0, 3):
    paperspace = doc.layouts.new(f'Page{i + 1}')
    paperspace.page_setup(size=(425, 297), margins=(1, 1, 1, 1), units="mm")
    viewport = paperspace.add_viewport(
        center=(212.5, 148.5),
        size=(425, 297),
        view_center_point=(start[0] + page_width  * i, start[1]),
        view_height=297,
        status=(i + 1),
    )
```
And Then, I can print a paperspace to pdf file using the way discussed in #1031. 

The only problem is I have to convert the dxf to multiple pdf files and merge them later.I am looking for a way to convert multiple paperspaces to only one pdf file.
The code is simple, I hope it can be useful to others.
Thanks again for ezdxf, a great library.
